config/crd/experimental/gateway.networking.k8s.io_udproutes.yaml

```diff
@@ -218,6 +218,98 @@ spec:
+                              type:
+                                allOf:
+                                - enum:
+                                  - IPAddress
+                                  - Hostname
+                                  - NamedAddress
+                                - enum:
+                                  - IPAddress
+                                  - NamedAddress
+                                default: IPAddress
+                                description: "Type of the address, either IPAddress
+                                  or NamedAddress. \n If NamedAddress is used this
+                                  is a custom and specific value for each implementation
+                                  to handle (and add validation for) according to
+                                  their own needs. \n For IPAddress the implementor
+                                  may expect either IPv4 or IPv6. \n Support: Core
+                                  (IPAddress) Support: Custom (NamedAddress)"
+                                type: string
+                              value:
+                                description: "Value of the address. The validity of
+                                  the values will depend on the type and support by
+                                  the controller. \n If implementations support proxy-protocol
+                                  (see: https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)
+                                  they must respect the connection metadata from proxy-protocol
+                                  in the match logic implemented for these address
+                                  values. \n Examples: `1.2.3.4`, `128::1`, `my-named-address`.
+                                  \n Support: Core"
+                                maxLength: 253
+                                minLength: 1
+                                type: string
+                            required:
+                            - value
+                            type: object
+                          type: array
+                        sourceAddresses:
+                          description: "SourceAddresses indicates the originating
+                            (source) network addresses which are valid for routing
+                            traffic. \n Support: Core"
+                          items:
+                            description: AddressMatch defines matching rules for network
+                              addresses by type.
+                            properties:
+                              type:
+                                allOf:
+                                - enum:
+                                  - IPAddress
+                                  - Hostname
+                                  - NamedAddress
+                                - enum:
+                                  - IPAddress
+                                  - NamedAddress
```

Comment:
Two enum sets. Something is wrong here.

Reply:
Yes, unfortunately re-using the `AddressType` from `Gateway` ended up with validation from the base type combining with the top-level validation. I didn't find a way to indicate to the generators to only use the highest level validation, so I ended up making a new type purpose-built to replace this: 05616c8daf9fbb4d48e7c2084a35ed609e3c5400